services/catalog/src/simcore_service_catalog/api/dependencies/services.py

```diff
@@ -88,28 +87,16 @@ async def get_service_from_registry(
     service_key: ServiceKey,
     service_version: ServiceVersion,
     director_client: Annotated[DirectorApi, Depends(get_director_api)],
-) -> ServiceGet:
+) -> ServiceMetaDataPublished:
     """
     Retrieves service metadata from the docker registry via the director
     """
     try:
-        if is_function_service(service_key):
-            frontend_service: dict[str, Any] = get_function_service(
-                key=service_key, version=service_version
-            )
-            _service_data = frontend_service
-        else:
-            # NOTE: raises HTTPException(status.HTTP_503_SERVICE_UNAVAILABLE) on ANY failure
-            services_in_registry = cast(
-                list[Any],
-                await director_client.get(
-                    f"/services/{urllib.parse.quote_plus(service_key)}/{service_version}"
-                ),
-            )
-            _service_data = services_in_registry[0]
-
-        service: ServiceGet = ServiceGet.parse_obj(_service_data)
-        return service
+        return await registry.get_registered_service(
+            service_key=service_key,
+            service_version=service_version,
+            director_client=director_client,
+        )
 
     except ValidationError as exc:
```

Comment:
Do you still need this try except block here?

Reply:
yes, here is where I convert lower level into HTTP exceptions